### DISCUSSION TITLE
DataTable is seemingly not amenable to summary-detail views - or if it is, how?

### DISCUSSION THREAD

**Discussion by vsajip** [Textualize/textual#1376] [Q&A]
A not uncommon application scenario is to manage a list of items in a table which shows some summary data about each item, and if you click on a row, then another pane populates with additional information for that item (which may include some of the summary information, but also includes other details which are too much to show in the table). Although `DataTable` would seem a good fit for this, it isn't really, because:

1. You can't select a whole row, only a cell. A row-select mode flag would be good to have.
2. You can't associate e.g. an object (or a unique object identifier) with a row in the table, to be used to populate the detail pane when that row (or something in it) is selected.

I've played with the new `ListView` functionality, but it seems to slow down a lot with big tables (of the order of 100K rows), whereas `DataTable` seems to proceed with async table population at a more reasonable pace. Have I missed some functionality in `DataTable` which allows it to be used in the above use case?

**Comment by watschft** [comment]
I have been able to figure out how to get the coordinates (row, column) of the selected cell:
        table = self.query_one("#summary_table")
        coords = table.cursor_cell
        rends = table._get_row_renderables(table.cursor_cell[0]) #gets the renderable for the row of the selected cell
        print(rends[0], rends[1], rends[2]) #you can access the cells in the row...
when you click on a cell you can set up an binding to 'enter' to an action to do the above.

My problem/request is that I can't figure out how to capture when a cell is selected by the user clicking on it.  I'd like to be able to get the selected cell by clicking on it like some other widgets do.  

I've experimented with a custom DataTable and capturing the on_click event with a little success, but not completely.  Maybe some pointers from the gurus could help us both.

**Reply by davep** [reply]
While I don't offer it as *the* solution to this (I suspect `DataTable` could do with some work to add things like this), I did write [this little example when someone asked a similar question on our Discord this morning](https://github.com/davep/textual-sandbox/blob/main/table_test.py).

Simply put: it subclasses `DataTable` and overrides `watch_cursor_cell` to track when the selected cell changes.

**Reply by watschft** [reply]
Thanks!  I’m sure there will be built in functionality for this sort of thing eventually.  I’m enjoying watching the project evolve!

One thing, however…  if you use arrow keys to move around the table, watch_cursor_cell fires every time.  If the reason for knowing the cell content is to kick off another query (more details about the stuff in the cell) that will result in lots of unnecessary queries… could it be made to only do it on ‘enter key’ or ‘click’?
